exporter/collector/logs.go

```diff
@@ -418,9 +418,9 @@ func (l logMapper) logToSplitEntries(
 		delete(attrsMap, SourceLocationAttributeKey)
 	}
 
-	// parse TraceSampled boolean from OTel attribute
-	if traceSampled, ok := attrsMap[TraceSampledAttributeKey]; ok {
-		entry.TraceSampled = traceSampled.Bool()
+	// parse TraceSampled boolean from OTel attribute or IsSampled OTLP flag
+	if traceSampled, ok := attrsMap[TraceSampledAttributeKey]; ok || logRecord.Flags().IsSampled() {
```

Comment:
Just curious, why did we have a "special" attribute in the first place?

Reply:
In the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/dc78006c12d9767fd2e35b691706c7572a76fd43/specification/logs/data-model-appendix.md#google-cloud-logging) we map all GCP LogEntry fields to a matching `gcp.*` attribute. [LogEntry](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry) has a field `traceSampled`, so we get `gcp.trace_sampled` as the matching attribute.

Reply:
Should we update the spec to say that we should use the log record sampled flag instead of this attribute?  We can keep it in the exporter for backwards-compatibility, but sampled isn't something I expect to get from attributes generally speaking.

Reply:
Yup good point, either way if we're adding support for log Flags we should update that spec doc to show it. I'll open a PR for that too

Reply:
https://github.com/open-telemetry/opentelemetry-specification/pull/3629